src/network/websocket/WebSocketServer.cpp

```diff
@@ -174,6 +179,7 @@ void SingleClientWSServer::onClose(connection_hdl hdl) {
 	LOG_F(INFO, "Server=%s, Endpoint=%s : Connection disconnected from %s", serverName.c_str(),
 		  path.c_str(), client.c_str());
 
+	std::lock_guard lock(protocolMapMutex);
```

Comment:
Genuine question, should we only lock the protocol map while accessing it, or should we keep it locked for the whole method? What do you think is there opportunity for race conditions if we do the former?

Reply:
I think it makes sense to only lock while accessing. Should I migrate to unique_lock and unlock post-access?

Reply:
I think if we used `unique_lock` we'd have a lot of code duplication since we'd have to this every time we access the map:
```c++
std::unique_lock lock(protocolMapMutex);
auto& protocolData = protocolMap.at(path);
lock.unlock();
```

Instead, it might be a bit cleaner to just make a new private method `ProtocolData& getProtocol(const std::string& protocolPath)` that handles synchronized access to the map, so it reduces code duplication, which would look something like:

```c++
ProtocolData& getProtocol(const std::string& protocolPath) {
    std::lock_guard lock(protocolMapMutex);
    return protocolMap.at(protocolPath);
}
```

Thoughts?